test/stdlib/StringViewHashing.swift

```diff
@@ -0,0 +1,47 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+// REQUIRES: reflection
+
+//
+// Tests for String view hashability
+//
+
+import StdlibUnittest
+
+#if _runtime(_ObjC)
+import Foundation
+#endif
+
+var StringViewHashingTests = TestSuite("StringViewHashingTests")
+
+let testString: String = "dogs"
```

Comment:
We should probably test a variety of strings. I would probably do something like…

```
let taggedNSString = "dog" as NSString //up to 11 ascii bytes can be stored in a tagged NSString
let asciiNSString = "The quick brown fox jumps over the lazy dog" as NSString
let nonAsciiNSString = "The quick brown fox jumps over the lazy dög" as NSString
let testStrings = ["" /* empty */, "dog" /* short ascii */, "dög" /* short non-ascii, "The quick brown fox jumps over the lazy dog" /* long ascii (long is defined as >15 bytes) */, "The quick brown fox jumps over the lazy dög" /* long non-ascii */, "👩‍👧" /* short multi-scalar */, "👩‍👩‍👧‍👧" /* long multi-scalar */, taggedNSString as String, asciiNSString as String, nonAsciiNSString as String]
```

and then just stick the existing tests in a loop over that array. It occurs to me that someone (probably me) should factor out a standard list of strings that covers all the likely edge cases so people can just reuse it.